action/lint.go

```diff
@@ -0,0 +1,171 @@
+}
+
+func directoryStructure(chartPath string) map[string]os.FileInfo {
+	structure := make(map[string]os.FileInfo)
+
+	chartInfo, err := os.Stat(chartPath)
+	if err != nil {
+		log.Die("Chart %s not found in workspace. Error: %v", chartPath, err)
+	}
+
+	if chartInfo.IsDir() {
+		files, _ := ioutil.ReadDir(chartPath)
+		for _, f := range files {
+			structure[f.Name()] = f
+		}
+	} else {
+		log.Die("Chart Path [%s] is not a directory.", chartPath)
+	}
+
+	return structure
+}
+
+func checkDirectoryStructure(structure map[string]os.FileInfo, chartPath string) {
+	if _, ok := structure["README.md"]; ok != true {
+		log.Warn("A README file was not found in %s", chartPath)
+	}
+
+	if _, ok := structure["Chart.yaml"]; ok != true {
+		log.Die("A Chart.yaml file was not found in %s", chartPath)
+	}
+
+	manifestInfo, ok := structure["manifests"]
+	if ok && manifestInfo.IsDir() {
+		// manifest files logic
+	} else {
+		log.Die("A manifests directory was not found in %s", chartPath)
+	}
+}
+
+// verifyMetadata checks the Chart.yaml file for a Name, Version, Description, and Maintainers
+func verifyMetadata(chartPath string) {
+	file := filepath.Join(chartPath, "Chart.yaml")
+	b, err := ioutil.ReadFile(file)
+	if err != nil {
+		log.Die("Error reading Chart.yaml.\nError: ", err)
+	}
+	var y *chart.Chartfile
+	if err = yaml.Unmarshal(b, &y); err != nil {
+		log.Die("Error parsing Chart.yaml file. \nError: ", err)
+	}
+	//require name, version, description, maintaners
+	if y.Name == "" {
+		log.Die("Missing Name specification in Chart.yaml file")
+	}
+	if y.Version == "" {
+		log.Die("Missing Version specification in Chart.yaml file")
+	}
+	if y.Description == "" {
+		log.Die("Missing description in Chart.yaml file")
+	}
+	if y.Maintainers == nil {
+		log.Die("Missing maintainers information in Chart.yaml file")
+	}
+}
+
+//@todo: refactor this method
+func verifyManifests(chartPath string) {
+	manifests, err := manifest.ParseDir(chartPath)
+	if err != nil {
+		log.Die("Error walking manifest files. Err: ", err)
+	}
+
+	for _, m := range manifests {
+		ref, _ := m.VersionedObject.Ref()
+		switch ref.Kind {
+		case "Pod":
+			p, _ := m.VersionedObject.Pod()
+			if p.Name == "" {
+				log.Die("Missing Name in %s", m.Source)
+			}
+
+			val, ok := p.Labels["heritage"]
+			if !ok || (val != "helm") {
+				log.Die("Missing a label: `heritage: helm` in %s", m.Source)
+			}
+
+		case "ReplicationController":
+			rc, _ := m.VersionedObject.RC()
+			if rc.Name == "" {
+				log.Die("Missing Name in %s", m.Source)
+			}
+
+			val, ok := rc.Labels["heritage"]
+			if !ok || (val != "helm") {
+				log.Die("Missing a label: `heritage: helm` in %s", m.Source)
+			}
+
+		case "Service":
+			s, _ := m.VersionedObject.Service()
+			if s.Name == "" {
+				log.Die("Missing Name in %s", m.Source)
+			}
+
+			val, ok := s.Labels["heritage"]
+			if !ok || (val != "helm") {
+				log.Die("Missing a label: `heritage: helm` in %s", m.Source)
+			}
```

Comment:
Probably if `Kind` is not one of the known kinds we should emit a warning (not an error though). `action.InstallOrder` is probable a cheap and easy way to see if this is a known Kind.